hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/FileGroupReaderBasedMergeHandle.java

```diff
@@ -139,23 +137,68 @@ private void init(CompactionOperation operation, String partitionPath, Option<Ho
 
       // Create the writer for writing the new version file
       fileWriter = HoodieFileWriterFactory.getFileWriter(instantTime, newFilePath, hoodieTable.getStorage(),
-          config, writeSchemaWithMetaFields, taskContextSupplier, getRecordType());
+          config, writeSchemaWithMetaFields, taskContextSupplier, recordType);
     } catch (IOException io) {
-      LOG.error("Error in update task at commit " + instantTime, io);
+      LOG.error("Error in update task at commit {}", instantTime, io);
       writeStatus.setGlobalError(io);
       throw new HoodieUpsertException("Failed to initialize HoodieUpdateHandle for FileId: " + fileId + " on commit "
           + instantTime + " on path " + hoodieTable.getMetaClient().getBasePath(), io);
     }
   }
 
-  protected abstract HoodieRecord.HoodieRecordType getRecordType();
-
   /**
    * Reads the file slice of a compaction operation using a file group reader,
    * by getting an iterator of the records; then writes the records to a new base file
    * using parquet writer.
    */
-  protected abstract void write();
+  public void write() {
```

Comment:
minor. 
lets fix the java docs for this method. it talks about parquet writer. 
but in case of metadata table, its not parquet.